src/Microsoft.DotNet.Interactive.HttpRequest.Tests/LexerTests.cs

```diff
@@ -0,0 +1,10 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.Interactive.HttpRequest.Tests;
+
+public class LexerTests
```

Comment:
We can leave out the empty test. It might not be necessary to directly test the lexer API if it's not public.

Reply:
I think we should add tests for the tokenization i.e., how we split whitespace, new lines, words and punctuation into tokens.

Reply:
We did discuss adding one to ensure the the same content (including different kinds of newlines) is parsed consistently so that we avoid OS-specific behaviors.

Beyond that, it's possible that the lexer tests will be redundant since this code is heavily exercised by the parser tests.

Reply:
They would still tightly pin-point the problem if something changed in how we tokenize text though, would it not (as opposed to tens of parser tests failing)?

Reply:
Absolutely. But I think those tests are still parser tests because the parser and nodes are the public API, which also surfaces the tokens. The lexer should not be part of the public API. It's an implementation detail of the parser and we don't have a use case for people to call it directly from outside, hence testing it directly should not be a goal.